bin/cron/update_dotd

```diff
@@ -0,0 +1,25 @@
+#!/usr/bin/env ruby
+require 'open-uri'
+require 'json'
+require_relative '../../deployment'
+
+slack_token = CDO.slack_token
+pagerduty_token = CDO.pagerduty_token
+channel_id = CDO.developers_channel_id
+
+# Get current developers room topic
+channel_info = open("https://slack.com/api/channels.info?token=#{slack_token}&channel=#{channel_id}").read
+current_topic = JSON.parse(channel_info)["channel"]["topic"]["value"]
+
+# Get DOTD on call from Pager Duty
+on_call = `curl -H "Content-type: application/json" -H "Authorization: Token token=#{pagerduty_token}" -X GET -G "https://codeorg.pagerduty.com/api/v1/users/on_call"`
+dotd_email = JSON.parse(on_call)["users"].first["email"]
+
+# Match Pager Duty email to Slack mention name
+mention_name = "DNE"
+users_list = open("https://slack.com/api/users.list?token=#{slack_token}").read
+JSON.parse(users_list)["members"].each { |member| mention_name = member["name"] if dotd_email.eql? member["profile"]["email"] }
```

Comment:
Oh, and LGTM :)
On Mar 29, 2016 20:30, "Laurel Fan" <laurel@code.org> wrote:

> Ah cool, I just didn't know what DNE stood for. I wouldn't put a ton of
> work into handling hypothetical edge cases; there are only like 15 possible
> values so we can handle those if/when they come up.
> On Mar 29, 2016 5:57 PM, "Tanya Parker" <notifications@github.com> wrote:
>
>> In bin/cron/update_dotd
>> <https://github.com/code-dot-org/code-dot-org/pull/7557#discussion_r57824790>
>> :
>>
>> > +slack_token = CDO.slack_token
>> > +pagerduty_token = CDO.pagerduty_token
>> > +channel_id = CDO.developers_channel_id
>> > +
>> > +# Get current developers room topic
>> > +channel_info = open("https://slack.com/api/channels.info?token=#{slack_token}&channel=#{channel_id}").read
>> > +current_topic = JSON.parse(channel_info)["channel"]["topic"]["value"]
>> > +
>> > +# Get DOTD on call from Pager Duty
>> > +on_call = `curl -H "Content-type: application/json" -H "Authorization: Token token=#{pagerduty_token}" -X GET -G "https://codeorg.pagerduty.com/api/v1/users/on_call"`
>> > +dotd_email = JSON.parse(on_call)["users"].first["email"]
>> > +
>> > +# Match Pager Duty email to Slack mention name
>> > +mention_name = "DNE"
>> > +users_list = open("https://slack.com/api/users.list?token=#{slack_token}").read
>> > +JSON.parse(users_list)["members"].each { |member| mention_name = member["name"] if dotd_email.eql? member["profile"]["email"] }
>>
>> I could explain it more and update the topic to something like "Couldn't
>> identify DotD, please check schedule."
>>
>> —
>> You are receiving this because you were assigned.
>> Reply to this email directly or view it on GitHub
>> <https://github.com/code-dot-org/code-dot-org/pull/7557/files/5e99bd51f041742d4d7392395139170035294ed4#r57824790>
>>
>